File: concrete/src/main/java/com/jaynewstrom/concrete/ConcreteWallDestructionAction.java
package com.jaynewstrom.concrete;

/**
 * Can be added to a {@link ConcreteWall} to be called when the wall is destroyed.
 */
public interface ConcreteWallDestructionAction<C> {
    /**
     * Called exactly once when the wall is destroyed.
     */
    void onWallDestroyed(C component);
}

File: concrete/src/main/java/com/jaynewstrom/concrete/ConcreteWallDestructionAction.kt
package com.jaynewstrom.concrete

/**
 * Can be added to a [ConcreteWall] to be called when the wall is destroyed.
 */
interface ConcreteWallDestructionAction<in C> {
    /**
     * Called exactly once when the wall is destroyed.
     */
    fun onWallDestroyed(component: C)
}
